guides/source/association_basics.md

```diff
@@ -2350,6 +2350,17 @@ end
 
 If a `before_add` callback throws an exception, the object does not get added to the collection. Similarly, if a `before_remove` callback throws an exception, the object does not get removed from the collection.
 
+NOTE: These callbacks are called only when the associated objects are added or removed through the association collection:
```

Comment:
@vipulnsward : `remove through the associated` sounds little incorrect to me. Maybe we could say `removed from the association collection.`

Changes: 
- `through` -> `from`
- Sentence should end with period here

Does that sound correct to you?

Reply:
It's correct as is. The idea is to communicate that adds/removals must go *through* the `has_many` accessor.